index.js

```diff
@@ -1,18 +1,27 @@
 const { codeFrameColumns } = require("@babel/code-frame");
 const Worker = require("jest-worker").default;
-const serialize = require("serialize-javascript");
+const { generate } = require("escodegen");
+const lave = require("lave");
 
 function terser(userOptions = {}) {
   if (userOptions.sourceMap != null) {
     throw Error("sourceMap option is removed, use sourcemap instead");
   }
 
-  const minifierOptions = serialize(
-    Object.assign({}, userOptions, {
-      sourceMap: userOptions.sourcemap !== false,
-      sourcemap: undefined,
-      numWorkers: undefined
-    })
+  const normalizedOptions = {
+    ...userOptions,
+    ...{ sourceMap: userOptions.sourcemap !== false }
+  };
+
+  for (let key of ["sourcemap", "numWorkers"]) {
```

Comment:
Unfortunately this is required because lave serializes `undefined` properties as well. And I assume terser also checks by Object.prototype.hasOwnProperty and therefore you get e.g. "sourcemap" is not an option error from within terser.